packages/pieces/community/openai/src/index.ts

```diff
@@ -23,21 +27,46 @@ Follow these instructions to get your OpenAI API Key:
 It is strongly recommended that you add your credit card information to your OpenAI account and upgrade to the paid plan **before** generating the API Key. This will help you prevent 429 errors.
 `;
 
-export const openaiAuth = PieceAuth.SecretText({
-  description: markdownDescription,
-  displayName: 'API Key',
+export const openaiAuth = PieceAuth.CustomAuth({
   required: true,
+  description: markdownDescription,
+  props: {
+    apiKey: PieceAuth.SecretText({
+      displayName: 'API Key',
+      required: true,
+    }),
+    baseUrl: Property.ShortText({
+      displayName: 'Base URL',
+      description: 'The base URL for the OpenAI instance.',
+      defaultValue: 'https://api.openai.com/v1',
+      required: true,
+    }),
+    apiVersion: Property.ShortText({
+      displayName: 'API Version',
+      description: 'The API version if you are using an Azure OpenAI resource',
+      required: false,
+    }),
+  },
   validate: async (auth) => {
     try {
+      
+      let headers;
+      if (auth.auth.apiVersion) {
+        headers = {
+          'api-key': auth.auth.apiKey,
+        };
+      } else {
+        headers = {
+          Authentication: `Bearer ${auth.auth.apiKey}`,
+        };
+      }
+      const baseUrl = auth.auth.baseUrl.replace(/\/$/, '') ?? 'https://api.openai.com/v1';
       await httpClient.sendRequest<{
         data: { id: string }[];
       }>({
-        url: 'https://api.openai.com/v1/models',
+        url: `${baseUrl}/models`,
```

Comment:
Hi,

The regular OpenAI no longer works, can you please check?

Reply:
Hi @abuaboud , it's fixed.